packages/xforms-engine/src/instance/SelectControl.ts

```diff
@@ -35,62 +34,72 @@ export type AnySelectDefinition = {
 	[V in ValueType]: SelectDefinition<V>;
 }[ValueType];
 
-interface SelectControlStateSpec<V extends ValueType> extends ValueNodeStateSpec<SelectValues<V>> {
+type AssertSupportedSelectValueType = (
```

Comment:
Question:

Is SelectNode the type definition and SelectControl the code to implement the functionality?

Reply:
Pretty close! `SelectNode` is the _client interface_, and `SelectControl` is the engine-internal implementation of it. Both are _also treated as types_, but only the former is exposed outside of the engine package.

Basically we have a "client interface" for each type of node. They intentionally hide a lot of details that are necessary for implementing the ODK XForms spec semantics (and underlying XPath spec semantics along with that). Those details are neither necessary nor appropriate for clients to access, and basically give us a clear boundary for engine internals which are fair game to change without needing to make any corresponding client changes.